src/render.rs

```diff
@@ -135,14 +186,33 @@ impl<'t, 'py> IntoOwnedContent<'t, 'py> for String {
     }
 }
 
-pub trait Render {
+/// Trait for resolving a template element into content suitable for
+/// further processing by another template element.
+pub trait Resolve {
     fn resolve<'t, 'py>(
         &self,
         py: Python<'py>,
         template: TemplateString<'t>,
         context: &mut Context,
     ) -> TemplateResult<'t, 'py>;
```

Comment:
I'm wondering if we shouldn't have the type (alias) be `ResolvedResult` here. Whereas for the `Render` trait and the `render` method result be `TemplateResult`, or perhaps `RenderedResult`.